packages/back-end/test/integrations/integration-query-generator.ts

```diff
@@ -0,0 +1,197 @@
+import {
+  DataSourceBase,
+  DataSourceInterface,
+  DataSourceType,
+} from "../../types/datasource";
+import {
+  AttributionModel,
+  ExperimentInterface,
+  ExperimentPhase,
+  Variation,
+} from "../../types/experiment";
+import { MetricInterface, MetricType } from "../../types/metric";
+import { getSourceIntegrationObject } from "../../src/services/datasource";
+
+const currentDate = new Date();
+
+// import experimentOverrides
+import experimentOverridesData from "./experiments.json";
+// TODO fix interface below
+interface ExperimentOverrideInterface {
+  id: string;
+  conversionWindowHours: number;
+  conversionDelayHour: number;
+  dimension?: string;
+  attributionModel: AttributionModel;
+}
+const experimentOverrides: ExperimentOverrideInterface[] = experimentOverridesData as ExperimentOverrideInterface[];
+
+// import metricOverrides
+import metricOverridesData from "./metrics.json";
+interface MetricOverrideInterface {
+  id: string;
+  type: MetricType;
+  ignoreNulls: boolean;
+  sql: string;
+  denominator?: string;
+}
+const metricOverrides: MetricOverrideInterface[] = metricOverridesData as MetricOverrideInterface[];
+
+// TODO: list of all db engines
+const engines: DataSourceType[] = [
+  "bigquery",
+  "postgres",
+  "snowflake",
+  "redshift",
+  "athena",
+  "mysql",
+  "mssql",
+  "clickhouse",
+];
+
+const baseExperimentPhase: ExperimentPhase = {
+  dateStarted: currentDate, // TODO: change date
+  dateEnded: currentDate, // TODO: change date
+  phase: "main",
+  reason: "",
+  coverage: 1,
+  variationWeights: [0.34, 0.33, 0.33],
+};
+
+// Pseudo-ExperimentInterface, missing following fields:
+// id: string;
+//  metricOverrides?: MetricOverride[];
+//  guardrails?: string[];
+//  activationMetric?: string;
+//  removeMultipleExposures?: boolean;
+//  segment?: string
+//  queryFilter?: string
+//  attributionModel?: AttributionModel;
+// And we have "dimensions" which isn't a field in ExperimentInterface, but
+// we're going to add it anyways
+const baseExperiment = {
```

Comment:
I would probably just add `id` here as an empty string.  That way this will be an actual ExperimentInterface object (since all the other missing properties are optional).